src/ios-deploy/ios-deploy.m

```diff
@@ -558,12 +559,18 @@ void mount_developer_image(AMDeviceRef device) {
     } else if (result == 0xe8000076 /* already mounted */) {
         NSLogOut(@"[ 95%%] Developer disk image already mounted");
     } else {
-        if (result == 0xe80000e2 /* device locked */) {
-            NSLogOut(@"The device is locked.");
+        if (result != 0) {
+            const char* msg = get_error_message(result);
+            NSString *description = @"unknown.";
+            if (msg) {
+                description = [NSString stringWithUTF8String:msg];
+                NSLogOut(@"Error: %@", description);
+            }
             NSLogJSON(@{@"Event": @"Error",
-                        @"Status": @"DeviceLocked"
-                        });
+                        @"Code": @(result),
+                        @"Status": description});
         }
+        
         on_error(@"Unable to mount developer disk image. (%x)", result);
```

Comment:
This is mostly duplicative now... but I see that people are string-matching this line in other programs so leaving it.